lib/ansible/config/manager.py

```diff
@@ -536,11 +537,20 @@ def get_config_value_and_origin(self, config, cfile=None, plugin_type=None, plug
                             raise AnsibleError("No setting was provided for required configuration %s" %
                                                to_native(_get_entry(plugin_type, plugin_name, config)))
                     else:
-                        value = defs[config].get('default')
                         origin = 'default'
-                        # skip typing as this is a templated default that will be resolved later in constants, which has needed vars
-                        if plugin_type is None and isinstance(value, string_types) and (value.startswith('{{') and value.endswith('}}')):
-                            return value, origin
+                        value = defs[config].get('default')
+                        if isinstance(value, string_types) and (value.startswith('{{') and value.endswith('}}')) and variables is not None:
+                            # template default values if possible
+                            # NOTE: cannot use is_template due to circular dep
+                            try:
+                                t = Template(value)
```

Comment:
I know this is just a copy&paste from elsewhere but I suspect we can simplify this and remove the call to `literal_eval` just by using `NativeTemplate` that would do the conversion instead.

Reply:
I'll mark that as a possible improvement, since that seems to require setting up my own environment and other classes also.

Reply:
Should be enough to just do:

```
from jinja2.nativetypes import NativeEnvironment
t = NativeEnvironment().from_string(value)
value = t.render(variables)
```

Reply:
not sure i see a change, but yes, this depends on order and since 3.6 was min version we relied on it keeping dictionary order.